traitsui/testing/tester/_ui_tester_registry/qt4/_interaction_helpers.py

```diff
@@ -205,6 +205,127 @@ def mouse_click_item_view(model, view, index, delay):
     )
 
 
+def mouse_dclick_item_view(model, view, index, delay):
+    """ Perform mouse double click on the given QAbstractItemModel (model) and
+    QAbstractItemView (view) with the given row and column.
+
+    Parameters
+    ----------
+    model : QAbstractItemModel
+        Model from which QModelIndex will be obtained
+    view : QAbstractItemView
+        View from which the widget identified by the index will be
+        found and key sequence be performed.
+    index : QModelIndex
+
+    Raises
+    ------
+    LookupError
+        If the index cannot be located.
+        Note that the index error provides more
+    """
+    check_q_model_index_valid(index)
+    rect = view.visualRect(index)
+    QTest.mouseDClick(
+        view.viewport(),
+        QtCore.Qt.LeftButton,
+        QtCore.Qt.NoModifier,
+        rect.center(),
+        delay=delay,
+    )
+
+
+def key_sequence_item_view(model, view, index, sequence, delay=0):
+    """ Perform mouse click on the given QAbstractItemModel (model) and
+    QAbstractItemView (view) with the given row and column.
+
+    Parameters
+    ----------
+    model : QAbstractItemModel
+        Model from which QModelIndex will be obtained
+    view : QAbstractItemView
+        View from which the widget identified by the index will be
+        found and key sequence be performed.
+    index : QModelIndex
+    sequence : str
+        Sequence of characters to be inserted to the widget identifed
+        by the row and column.
+
+    Raises
+    ------
+    Disabled
+        If the widget cannot be edited.
+    LookupError
+        If the index cannot be located.
+        Note that the index error provides more
+    """
+    check_q_model_index_valid(index)
+    widget = view.indexWidget(index)
+    if widget is None:
+        raise Disabled(
+            "No editable widget for item at row {!r} and column {!r}".format(
+                index.row(), index.column()
+            )
+        )
+    QTest.keyClicks(widget, sequence, delay=delay)
```

Comment:
Perhaps we need to call `check_key_compat` here like we do in `key_sequence_textbox`?  I think I recall this being to keep features the same with wx (so tests can remain agnostic).

For TraitsUI tests this is good and make sense.  However, a downstream user may only care about qt so the features could be nice.  I believe qt could do things like `\b` for backspace whereas to preserve wx/qt compat we now use "Backspace", etc.

Reply:
Given that this support is now going to be on qt any way this concern doesn't really seem important